google/cloud/bigtable/client.py

```diff
@@ -94,8 +310,40 @@ def __init__(
         instance_id: str,
         table_id: str,
         app_profile_id: str | None = None,
+        metadata: list[tuple[str, str]] | None = None,
     ):
-        raise NotImplementedError
+        """
+        Initialize a Table instance
+
+        Must be created within an async run loop context
+
+        Args:
+            instance_id: The Bigtable instance ID to associate with this client
+                instance_id is combined with the client's project to fully
+                specify the instance
+            table_id: The ID of the table.
+            app_profile_id: (Optional) The app profile to associate with requests.
+                https://cloud.google.com/bigtable/docs/app-profiles
+            metadata: a list of metadata headers to be attached to all calls with this client
+        Raises:
+          - RuntimeError if called outside of an async run loop context
+        """
+        self.client = client
+        self.instance = instance_id
+        self.table_id = table_id
+        self.app_profile_id = app_profile_id
+        self.metadata = metadata or []
+        # raises RuntimeError if called outside of an async run loop context
+        try:
+            self._register_instance_task = asyncio.create_task(
+                self.client.register_instance(instance_id)
```

Comment:
should there a deregister call as well?

Reply:
I'm not sure I understand. This task is needed because we can't call await from `__init__`, so we schedule it as a background task instead